src/EditorFeatures/Core/Implementation/NavigationBar/NavigationBarController_ModelComputation.cs

```diff
@@ -81,45 +77,60 @@ private void StartSelectedItemUpdateTask()
 
         private async ValueTask SelectItemAsync(CancellationToken cancellationToken)
         {
-            // Switch to the UI so we can determine where the user is.
+            // Switch to the UI so we can determine where the user is and determine the state the last time we updated
+            // the UI.
             await this.ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
             var currentView = _presenter.TryGetCurrentView();
             var caretPosition = currentView?.GetCaretPoint(_subjectBuffer);
             if (!caretPosition.HasValue)
                 return;
 
-            // Ensure the latest model is computed and grab it while on the UI thread.
-            var model = await _computeModelQueue.WaitUntilCurrentBatchCompletesAsync().ConfigureAwait(true);
+            var position = caretPosition.Value.Position;
+            var lastPresentedInfo = _lastPresentedInfo;
 
             // Jump back to the BG to do any expensive work walking the entire model
             await TaskScheduler.Default;
 
-            var currentSelectedItem = ComputeSelectedTypeAndMember(model, caretPosition.Value, cancellationToken);
+            // Ensure the latest model is computed.
+            var model = await _computeModelQueue.WaitUntilCurrentBatchCompletesAsync().ConfigureAwait(true);
+
+            var currentSelectedItem = ComputeSelectedTypeAndMember(model, position, cancellationToken);
+
+            GetProjectItems(out var projectItems, out var selectedProjectItem);
+            if (Equals(model, lastPresentedInfo.model) &&
+                Equals(currentSelectedItem, lastPresentedInfo.selectedInfo) &&
+                Equals(selectedProjectItem, lastPresentedInfo.selectedProjectItem) &&
+                projectItems.SequenceEqual(lastPresentedInfo.projectItems))
+            {
+                // Nothing changed, so we can skip presenting these items.
+                return;
+            }
```

Comment:
we can do all this comparison work in the BG now.  also, the equality checks actually work now (they were broken before as they only checked referential identity).